src/Sylius/Bundle/SettingsBundle/Resolver/DefaultSettingsResolver.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Bundle\SettingsBundle\Resolver;
+
+use Doctrine\ORM\NonUniqueResultException;
+use Sylius\Component\Resource\Repository\RepositoryInterface;
+
+/**
+ * Default implementation that resolves settings in the simplest way possible. It tries to find a schema by it's name
+ * and optionally the given namespace.
+ *
+ * @author Steffen Brem <steffenbrem@gmail.com>
+ */
+class DefaultSettingsResolver implements SettingsResolverInterface
+{
+    /**
+     * @var RepositoryInterface
+     */
+    protected $settingRepository;
+
+    /**
+     * @param RepositoryInterface $settingRepository
+     */
+    public function __construct(RepositoryInterface $settingRepository)
+    {
+        $this->settingRepository = $settingRepository;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function resolve($schemaAlias, $namespace = null)
+    {
+        try {
+            $criteria = [
+                'schemaAlias' => $schemaAlias,
+            ];
+
+            if (null !== $namespace) {
+                $criteria['namespace'] = $namespace;
+            }
+
+            return $this->settingRepository->findOneBy($criteria);
+        } catch (NonUniqueResultException $e) {
```

Comment:
I wil remove that try block. Since the namespace and schema alias already have a unique constraint I only need to properly send the query containing the schema and the namespace. Right now it skips the namespace criteria when it's null, but I think that is wrong and we need to always send it.